### ISSUE TITLE
OGRVRTLayer throws CE_Failure with description ignoring it

### ISSUE THREAD

**Issue by tcommandeur** [OSGeo/gdal#11518]
### What is the bug?

I'm using Mapserver (v8.2.2) and a VRT file with a SrcRegion defined. And when performing a WMS getfeatureinfo request the request fails. To my opinion this is due to the type of error thrown in `ogrvrtlayer.cpp`.

Case:
I'm doing a WMS request that results in an error thrown by the OGR VRT layer: `CPLError(CE_Failure, CPLE_AppDefined,"Spatial filter should be polygon when a SrcRegion is defined. Ignoring it");`. However the `poSrcRegion` is correctly set and execution continues.

This exception is caught by mapserver with `if (CPLGetLastErrorType() == CE_Failure)` that results in failure.

Expected result:
OGR VRT layer to throw a CE_Warning like this: `CPLError(CE_Warning, CPLE_AppDefined,"Spatial filter should be polygon when a SrcRegion is defined. Ignoring it");`

### Steps to reproduce the issue

**Dataset:**
BAG geopackage: https://service.pdok.nl/lv/bag/atom/bag.xml

**VRT file:**
```
<OGRVRTDataSource>
    <OGRVRTLayer name="pand">
        <SrcDataSource>bag-light.gpkg</SrcDataSource>
        <SrcLayer>pand</SrcLayer>
        <LayerSRS>EPSG:28992</LayerSRS>
        <GeometryField>
            <!-- Clip is only implemented at GeometryField level and not at OGRVRTLayer level -->
            <SrcRegion clip="false">POLYGON((120000 435000, 155000 435000, 155000 470000, 120000 470000, 120000 435000))</SrcRegion>
        </GeometryField>
    </OGRVRTLayer>
</OGRVRTDataSource>
```

No idea how to reproduce this with ogr2ogr since I cannot find out how the `m_poFilterGeom` is set that seems to be something different then a polygon.

### Versions and provenance

GDAL 3.6, 3.9.3, 3.10.0 self-built from binaries.

**Comment by tcommandeur** [comment]
I tested the PR and the results are as expected. Thanks for the incredible quick fix. 
Looking at the label does it means we can expect this fix to be released in GDAL 3.10.1?

**Comment by rouault** [comment]
> Looking at the label does it means we can expect this fix to be released in GDAL 3.10.1?

yes
